man/man1/FeynHiggs.1
.TH FEYNHIGGS 1 "30-Jan-2018"
.SH NAME
.PP
FeynHiggs \- the FeynHiggs command-line frontend
.SH SYNOPSIS
FeynHiggs parameterfile [flags]
.SH DESCRIPTION
FeynHiggs calculates the masses and couplings of the Higgs sector in the
MSSM.  FeynHiggs provides a convenient way of invoking the subroutines
of the FeynHiggs library from the command line.  The input parameters are
read from a file and the output is displayed in a human-readable form on
screen, but can be piped through a utility program to yield
machine-readable data files, e.g. for plotting.
.SH PARAMETER FILES
The input parameters for FeynHiggs are read from a file.  FeynHiggs
attempts to read an SLHA file first and if that fails, falls back onto
its native format (description follows below).  Output-wise, the default
is to write out an SLHA file (original filename + .fh-NNN) if the input
was one, and write the output to screen otherwise.  Output style can be
changed by appending #SLHA to the filename to force SLHA output, and
#FH to force screen output.

Description of the native FeynHiggs format: comments begin with a #
and continue to the end of the line.  Apart from a possible comment,
each line has the format
.P
  var  lower  [upper]  [*][/][step]
.P
and declares a loop over
.I var
from
.I lower
to
.I upper
in steps of
.I step.
A "*" in front of
.I step
results in a logarithmic stepping, i.e.
.I var
is multiplied by
.I step
in each iteration.  A prefix of "/" indicates that
.I step
is to be interpreted as the number of steps, rather than the step size.
.P
Simple assignments of the form
.I name lower
(i.e. no loop) are just a special case of this since the items in
brackets may be omitted.
.P
.I var
is the name of the parameter, which is one of
.br
-------------------+--------------------------------------
.br
invAlfa0,          | the inverse of the electromagnetic
.br
invAlfaMZ          | coupling constant at 0, MZ
.br
-------------------+--------------------------------------
.br
AlfasMZ            | the strong coupling constant at MZ
.br
-------------------+--------------------------------------
.br
GF                 | the Fermi constant
.br
-------------------+--------------------------------------
.br
MS, MC, MT, MB     | the strange, charm, top, bottom masses
.br
-------------------+--------------------------------------
.br
MW, MZ             | the W and Z masses
.br
-------------------+--------------------------------------
.br
TB                 | tan(beta)
.br
-------------------+--------------------------------------
.br
MA0, MHp           | the CP-odd and charged Higgs mass
.br
                   | (only one should be specified)
.br
-------------------+--------------------------------------
.br
MSusy              | a generic soft-SUSY breaking mass
.br
                   | used as a default value for all
.br
                   | MgSL, MgSE, MgSQ, MgSU, MgSD.
.br
-------------------+--------------------------------------
.br
MgSL, MgSQ         | individual soft-SUSY breaking masses
.br
MgSE, MgSU, MgSD   | for the slepton and squark doublets
.br
g = 1..3           | and singlets.
.br
                   | Default: all MgSx = MSusy.
.br
-------------------+--------------------------------------
.br
Abs(MUE), Re(MUE)  | the Higgs mixing parameter
.br
Arg(MUE), Im(MUE)  |
.br
-------------------+--------------------------------------
.br
Abs(Af), Re(Af)    | the sfermion mixing parameters
.br
Arg(Af), Im(Af)    |
.br
f = e,mu,tau,      |
.br
    u,c,t,         |
.br
    d,s,b          |
.br
-------------------+--------------------------------------
.br
Abs(Xf), Re(Xf)    | alternative sfermion mixing parameter;
.br
Arg(Xf), Im(Xf)    | since Xf and Af are not independent,
.br
f = tau,t,b        | only one should be specified
.br
-------------------+--------------------------------------
.br
Abs(M_1), Re(M_1)  | the U1 gaugino mass parameter
.br
Arg(M_1), Im(M_1)  |
.br
-------------------+--------------------------------------
.br
Abs(M_2), Re(M_2)  | the SU2 gaugino mass parameter
.br
Arg(M_2), Im(M_2)  |
.br
-------------------+--------------------------------------
.br
Abs(M_3), Re(M_3)  | the gluino mass parameter
.br
Arg(M_3), Im(M_3)  |
.br
-------------------+--------------------------------------
.br
Qtau, Qt, Qb       | the scales at which the stau, stop,
.br
                   | and sbottom DRbar masses are defined
.br
                   | Default: all 0 (= on-shell masses)
.br
-------------------+--------------------------------------
.br
CKMlambda          | the CKM parameters
.br
CKMA               |
.br
CKMrhobar          |
.br
CKMetabar          | Default: all -1 (= use PDG values)
.br
-------------------+--------------------------------------
.br
Abs(deltaFXYij)    | the flavour-violating
.br
Arg(deltaFXYij)    | parameters
.br
Re(deltaFXYij)     |
.br
Im(deltaFXYij)     |
.br
F = L,E,Q,U,D      |
.br
XY = LL,LR,RL,RR   |
.br
ij = 12,23,13,     | Default: all 0
.br
-------------------+--------------------------------------
.br
scalefactor        | the ren. scale is MT*scalefactor
.br
-------------------+--------------------------------------
.br
prodSqrts          | sqrt(s) for the production x-sections
.br
-------------------+--------------------------------------
.P
Needless to add that complex quantities should be specified EITHER
through Abs and Arg OR through Re and Im.  In conflicting
cases a warning is issued and the Abs/Arg value takes
precedence.
.SH FLAGS
The flags are given as an optional 8-digit string on the command line,
where the default is 42020110.  The positions in the string are as
follows:
.TP
#1: mssmpart = 0..4
specifies the scope of the 1-loop part:
.br
0: MT^4-approximation
.br
1: top/stop-sector
.br
2: top/stop- + bottom/sbottom-sector
.br
3: full (s-)quark/lepton-sector
.br
4: full MSSM
.br
.B recommended: 4
.TP
#2: higgsmix = 1..3
determines the mixing in the Higgs sector:
.br
1: All non-diagonal (ND) Self Energies = 0
.br
2: All CP-violating ND Self Energies = 0
.br
3: Full 3x3 mixing in the neutral sector
.br
.B This means:
.br
.B 2: evaluation in the rMSSM
.br
.B 3: evaluation in the cMSSM
.TP
#3: p2approx = 0..4
determines the approximation for the 1-loop result:
.br
0: None, i.e. full determination of the propagator matrices's poles,
   UHiggs is evaluated at p^2 = m^2
.br
1: p^2 = 0 approximation
.br
2: All self energies are calculated at p^2 = m_tree^2
.br
3: imaginary parts of self-energies are discarded
.br
4: as 0, but with UHiggs evaluated at p^2 = 0
.br
.B recommended: 4
.TP
#4: looplevel = 0..2
determines the inclusion of higher-order corrections:
.br
0: tree level
.br
1: one-loop contributions only
.br
2: include various two-loop contributions
.br
.B recommended: 2
.TP
#5: loglevel = 0..3
determines the inclusion of log resummations:
.br
0: no log resummation
.br
1: NLL resummation (for large MSUSY)
.br
2: NLL resummation (for large MCha,MNeu,MGlu,MSUSY)
.br
3: NNLL resummation (for large MCha,MNeu,MGlu,MSUSY)
.br
.B recommended: 3
.TP
#6: runningMT = 0..3
determines which top mass shall be used in the 1-/2-loop corrections:
.br
0: pole MT
.br
1: running MT (SM MSbar 2L)
.br
2: running MT (SM MSbar 1L)
.br
3: running MT (MSSM DRbar)
.br
.B recommended: 1
.TP
#7: botResum = 0..1
determines whether the O(tan beta^n) corrections shall be resummed:
.br
0: no resummation
.br
1: 1L resummation
.br
2: 1L resummation for Higgs self-energies, 2L resummation for Higgs decays
.br
.B recommended: 2
.TP
#8: tlCplxApprox = 0..3
determines how the two-loop corrections are treated in the presence of
complex parameters (cMSSM):
.br
0: all corrections (asat, asab, atat, atab) are computed in
   the rMSSM
.br
1: only the cMSSM asat corrections are used
.br
2: the cMSSM asat corrections are combined with the remaining
   corrections in the rMSSM
.br
3: the cMSSM a_s a_t corrections are combined with the remaining
   corrections, interpolated in the complex phases (cf. FHSelectIPol)
.SH EXAMPLE
Using the example parameter file that comes with FeynHiggs, the
command
.P
  FeynHiggs example/var.in
.P
should result in the following Higgs masses and mixings:
.P
| Mh0           =     113.89008624
.br
| MHH           =     195.87658697
.br
| MA0           =     200.00000000
.br
| MHp           =     213.63955069
.br
| SAeff         =    -0.35592463    0.00000000
.br
| UHiggsRe      =     0.99672973    0.08080741    0.00000000
.br
|                    -0.08080741    0.99672973    0.00000000
.br
|                     0.00000000    0.00000000    1.00000000
.P
and the following widths and branching ratios:
.P
| h0-b-b        =    6.018044E-03   0.825117       0.725474
.br
| HH-b-b        =    5.165486E-02   0.559950       2.526224E-03
.br
| A0-b-b        =    5.896187E-02   0.848461       2.280889E-03
.br
| Hp-t-b        =    6.224386E-02   0.781530
.P
Before the results, FeynHiggs lists the parameters of the calculation.
.P
With all the detailed information on widths and branching ratios, the
screen can become rather crowded.  To mask off this detailed output,
type
.P
  FeynHiggs ... | grep -v %
.SH CONVERSION TO MACHINE-READABLE OUTPUT
The output of FeynHiggs can be converted into a format more amenable to
further processing, e.g. plotting, by the "table" utility.  This works as
follows:
.P
  FeynHiggs ... | table TB Mh0 > var.out
.P
var.out then contains two columns listing TB and Mh0 for each pass of
the loop.
.SH SPS BENCHMARK SCENARIOS
The SPS benchmark scenarios are predefined.  The input files can be
found in example/SPS/.  To run a scenario, type e.g.
.P
  FeynHiggs example/SPS/SPS1a.in
.P
For more information on the SPS, see
.P
  http://www.ippp.dur.ac.uk/~georg/sps/
.SH LES HOUCHES BENCHMARK SCENARIOS
The Les Houches benchmark scenarios for Higgs boson searches at hadron
colliders are predefined.  The input files can be found in
example/LHBMS/.  To run a scenario, type e.g.
.P
  FeynHiggs example/SPS/mhmax.in
.P
For more information on the Les Houches Benchmark scenarios, see
Eur. Phys. J. C26 (2003) 601 [hep-ph/0202167].
.SH SUSY LES HOUCHES ACCORD DATA
FeynHiggs can handle data files in the SUSY Les Houches Accord (SLHA)
format.  If the input file is in SLHA format, the results (masses,
widths, etc.) are equally written out in SLHA format, to a file named as
the source file suffixed by ".fh".  For example:
.P
  FeynHiggs example/SLHA/TestEXTPAR.spc
.P
produces example/SLHA/TestEXTPAR.spc.fh.  FeynHiggs can process SLHA
files in which the input parameters are given (Block EXTPAR) as well as
files in which the masses and mixings is given (Blocks MASS, STOPMIX,
etc.), where in the latter case the input parameters are reconstructed
from the mass spectrum.  For instance, try:
.P
  FeynHiggs example/SLHA/SphenoRP.spc
.P
For more information on the SLHA, see
.P
  http://home.fnal.gov/~skands/slha/
.SH ENVIRONMENT
The FeynHiggs debug level can be set using the FHDEBUG environment
variable, e.g.
.P
  setenv FHDEBUG 1   (in tcsh)
.br
  export FHDEBUG=1   (in bash)
.P
The pager spawned to view FeynHiggs output is taken from the environment
variable PAGER and defaults to "less" if PAGER is not set.  Should a
program abort occur with FeynHiggs, it is advisable to set PAGER to an
empty string, e.g.
.P
  setenv PAGER
.P
This prevents a pager from being started and ensures that the output
is shown right up to the point where the abort occurs.
.SH SEE ALSO
.PP
libFH(1), MFeynHiggs(1)


src/Frontend/FeynHiggs.F
* FeynHiggs.F
* the command-line front end for FeynHiggs
* this file is part of FeynHiggs
* last modified 8 Jan 21 th

#include "FHCouplings.h"
#include "types.h"
#include "debug.h"
#include "version.h"
#include "flags.h"


	program FeynHiggs
	implicit none

#include "FHRecord.h"
#include "SLHA.h"

	character*128 argv
	integer argc, type, error, loop, keyslha, i
	integer mssmpart, higgsmix, p2approx, looplevel, loglevel
	integer runningMT, botResum, tlCplxApprox
	RecordDecl(record)
	ComplexType slhadata(nslhadata)

#if U77EXT
#define len_trim lnblnk
	integer iargc, lnblnk
	external iargc, lnblnk
#endif

	character*16 flags

	character*72
     &    dmssmpart(mssmpart_min-1:mssmpart_max+1),
     &    dhiggsmix(higgsmix_min-1:higgsmix_max+1),
     &    dp2approx(p2approx_min-1:p2approx_max+1),
     &    dlooplevel(looplevel_min-1:looplevel_max+1),
     &    dloglevel(loglevel_min-1:loglevel_max+1),
     &    drunningMT(runningMT_min-1:runningMT_max+1),
     &    dbotResum(botResum_min-1:botResum_max+1),
     &    dtlCplxApprox(tlCplxApprox_min-1:tlCplxApprox_max+1)

	data flags /defaultflags/

	data dmssmpart /
     &    "???",
     &    "MT^4 approximation (for real parameters only)",
     &    "top/stop sector",
     &    "top/stop + bottom/sbottom sector",
     &    "full (s)quark/(s)lepton sector",
     &    "full MSSM [recommended]",
     &    "???" /

	data dhiggsmix /
     &    "???",
     &    "no mixing in the neutral Higgs sector",
     &    "2x2 (h0-HH) mixing in the neutral Higgs sector",
     &    "3x3 (h0-HH-A0) mixing in the neutral Higgs sector",
     &    "4x4 (G0-h0-HH-A0) mixing [experimental]",
     &    "???" /

	data dp2approx /
     &    "???",
     &    "no approximation, UHiggs evaluated at p^2 = m^2",
     &    "p^2 = 0 approximation",
     &    "self-energies are calculated at p^2 = m_tree^2",
     &    "imaginary parts of self-energies are discarded",
     &    "no approx., UHiggs evaluated at p^2 = 0 [recommended]",
     &    "???" /

	data dlooplevel /
     &    "???",
     &    "tree level only",
     &    "include 1L corrections",
     &    "include 2L corrections",
     &    "???" /

	data dloglevel /
     &    "???",
     &    "no log resummation",
     &    "NLL resummation (for large MSUSY)",
     &    "NLL resummation (for large MCha,MNeu,MGlu,MSUSY)",
     &    "NNLL resummation (for large MCha,MNeu,MGlu,MSUSY)",
     &    "NNLL resummation (with eff. THDM in case of low MA)",
     &    "???" /

	data drunningMT /
     &    "???",
     &    "use pole top mass",
     &    "use running top mass (SM MSbar 2L)",
     &    "use running top mass (SM MSbar 1L)",
     &    "use running top mass (MSSM DRbar)",
     &    "???" /

	data dbotResum /
     &    "???",
     &    "non-resummed MB in 1L/2L Higgs self-energy corrections",
     &    "resummed 1L-MB in 1L/2L Higgs self-energy corrections",
     &    "resummed 1L-MB in Higgs se, 2L-MB in decays [rec.]",
     &    "???" /

	data dtlCplxApprox /
     &    "???",
     &    "all corrections evaluated in the rMSSM",
     &    "asat,atat in cMSSM only",
     &    "asat,atat in cMSSM, rest in rMSSM",
     &    "asat,atat in cMSSM, rest interpolated",
     &    "???" /

	call pageron

	argc = iargc()
	if( argc .lt. 1 .or. argc .gt. 3 ) goto 999

	if( argc .ge. 2 ) call getarg(2, flags)
	call FHSetFlagsString(error, flags)

	call FHRetrieveFlags(error,
     &    mssmpart, higgsmix, p2approx, looplevel, loglevel,
     &    runningMT, botResum, tlCplxApprox)

#define ShowFlag(t,f,lo,hi) t, f, d f(min(max(f,lo-1),hi+1))

101	format(A, " = ", I1, T20, "# ", A)
	print 101,
     &    ShowFlag("mssmpart",     mssmpart,     mssmpart_min, mssmpart_max),
     &    ShowFlag("higgsmix",     higgsmix,     higgsmix_min, higgsmix_max),
     &    ShowFlag("p2approx",     p2approx,     p2approx_min, p2approx_max),
     &    ShowFlag("looplevel",    looplevel,    looplevel_min, looplevel_max),
     &    ShowFlag("loglevel",     loglevel,     loglevel_min, loglevel_max),
     &    ShowFlag("runningMT",    runningMT,    runningMT_min, runningMT_max),
     &    ShowFlag("botResum",     botResum,     botResum_min, botResum_max),
     &    ShowFlag("tlCplxApprox", tlCplxApprox, tlCplxApprox_min, tlCplxApprox_max)

	if( error .ne. 0 ) stop

	call getarg(1, argv)
	i = index(argv, "#")
	type = -1
	if( i .gt. 0 ) then
	  type = 2
	  if( argv(i+1:) .eq. "SLHA" ) type = 0
	  argv(i:) = " "
	endif

	call FHReadRecord(error, record, slhadata, argv)
* 0 = SLHA, 2 = FH file, all others are true errors
	if( ibclr(error, 1) .ne. 0 ) goto 1

#if SLHAPARA
	keyslha = -2
#else
	keyslha = -4
#endif
	if( type .eq. 0 .and. error .ne. 0 ) then
	  call SLHAClear(slhadata)
	  keyslha = -1
	endif

	if( type .eq. -1 ) type = error

	if( type .eq. 0 ) then
	  i = 0
	  if( argv .ne. "-" ) i = len_trim(argv) + 1
	endif

* turn on verbose mode:
	record(iAdmin,iStep) = 1

	loop = 0

2	call FHLoopRecord(error, record)
c	if( error .eq. -2 ) goto 2
	if( error .ne. 0 ) goto 1

	loop = loop + 1

	call FHSetRecord(error, record)
	if( error .ne. 0 ) goto 2

	if( type .eq. 0 ) then
	  call FHOutputSLHA(error, slhadata, keyslha)
	  if( error .ne. 0 ) then
	    print *, "Error in FHOutputSLHA ", error
	  else
	    if( i .gt. 0 ) write(argv(i:),'(".fh-",I3.3)') loop
	    call SLHAWrite(error, slhadata, argv)
	  endif
	else
	  call FHOutput(error, "stdout", -1, record(iprodSqrts,iVar))
	  if( error .ne. 0 ) print *, "Error in FHOutput ", error
	  print 200
	endif

	goto 2


#define DESC_FMT (/I5,": ",A)

200	format(/"-------------------- END OF OUTPUT ---------------------")

100	format ("FeynHiggs ", FHVERSION
     &    /"built on ", FHBUILDDATE
     &    //"USAGE:  ", A, " <inputfile> [<flags>:", defaultflags, "]"
     &    //"<inputfile> is a file containing the parameters"
     &    //"<flags> is an 8-digit string with"
     &    /"  #1 = mssmpart = scope of the calculation",
     &    mssmpart_n DESC_FMT
     &    /"  #2 = higgsmix = mixing in the neutral Higgs sector",
     &    higgsmix_n DESC_FMT
     &    /"  #3 = p2approx = approximation for the 1L result",
     &    p2approx_n DESC_FMT
     &    /"  #4 = looplevel, inclusion of higher-order corrections",
     &    looplevel_n DESC_FMT
     &    /"  #5 = loglevel, inclusion of log resummations",
     &    loglevel_n DESC_FMT
     &    /"  #6 = runningMT, pole vs. running MT",
     &    runningMT_n DESC_FMT
     &    /"  #7 = botResum, MB resummation",
     &    botResum_n DESC_FMT
     &    /"  #8 = tlCplxApprox, complex parameters in the 2L corrections",
     &    tlCplxApprox_n DESC_FMT
     &    /)

999	call getarg(0, argv)
	print 100, Strip(argv),
     &    (i, dmssmpart(i),     i = mssmpart_min, mssmpart_max),
     &    (i, dhiggsmix(i),     i = higgsmix_min, higgsmix_max),
     &    (i, dp2approx(i),     i = p2approx_min, p2approx_max),
     &    (i, dlooplevel(i),    i = looplevel_min, looplevel_max),
     &    (i, dloglevel(i),     i = loglevel_min, loglevel_max),
     &    (i, drunningMT(i),    i = runningMT_min, runningMT_max),
     &    (i, dbotResum(i),     i = botResum_min, botResum_max),
     &    (i, dtlCplxApprox(i), i = tlCplxApprox_min, tlCplxApprox_max)

1	call flush(6)
	call pageroff
	end
